dns/name.py

```diff
@@ -1082,3 +1121,178 @@ def from_wire(message: bytes, current: int) -> Tuple[Name, int]:
     parser = dns.wire.Parser(message, current)
     name = from_wire_parser(parser)
     return (name, parser.current - current)
+
+
+# RFC 4471
+
+_MINIMAL_OCTET = b"\x00"
+_MINIMAL_OCTET_VALUE = 0x00
+_SUCCESSOR_PREFIX = Name([_MINIMAL_OCTET])
+_MAXIMAL_OCTET = b"\xff"
+_MAXIMAL_OCTET_VALUE = 0xFF
+_AT_SIGN_VALUE = 0x40
+_LEFT_SQUARE_BRACKET_VALUE = 0x5B
+
+
+def _wire_length(labels):
+    return functools.reduce(lambda v, x: v + len(x) + 1, labels, 0)
+
+
+def _pad_to_max_name(name):
+    needed = 255 - _wire_length(name.labels)
+    new_labels = []
+    while needed > 0:
+        if needed > 64:
+            new_labels.append(_MAXIMAL_OCTET * 63)
+            needed -= 64
+        elif needed > 2:
+            new_labels.append(_MAXIMAL_OCTET * (needed - 1))
+            needed = 0
+        else:
+            # Can't add just one byte as a new label.
+            needed = 0
```

Comment:
Couldn't this be a `while needed > 64` loop and an `if needed > 2` loop, rather than nested loops?

Also, the `Can't add just one byte as a new label.` comment is confusing, as it could enter that block with `needed == 2`.  Should the `> 2` earlier be ` >= 2`?

Reply:
Simplified!